### PR TITLE
Make Error Message more descriptive.

### LINKED ISSUE #5684: Ticketing not enabled error message
**Describe the bug**
Ticketing not enabled error message comes when an unverified user tries to register an event or someone tries to register the event without giving a location.

**Expected behavior**
Error Message should be more clear and helpful to the user.

**Additional context**
Working on it.

### PR DESCRIPTION
**Pull request by prateekj117**
Fixes #5684 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/events.py [hunk 1]
```diff
@@ -118,8 +118,12 @@ def before_post(self, args, kwargs, data=None):
         user = User.query.filter_by(id=kwargs['user_id']).first()
         modules = Module.query.first()
         if data.get('is_ticketing_enabled', False) and not modules.ticket_include:
-            raise ForbiddenException({'source': '/data/attributes/is-ticketing-enabled'},
-                                     "Ticketing is not enabled in the system")
+            if user.is_verified:
```
Comment (iamareebjamal) [inline app/api/events.py]: These conditions are in wrong place. They have nothing to do with tickets module. Add them as separate conditions

File: app/api/events.py [hunk 2]
```diff
@@ -452,10 +456,15 @@ def before_patch(self, args, kwargs, data=None):
         :param data:
         :return:
         """
+        user = User.query.filter_by(id=kwargs['user_id']).first()
         modules = Module.query.first()
         if data.get('is_ticketing_enabled', False) and not modules.ticket_include:
-            raise ForbiddenException({'source': '/data/attributes/is-ticketing-enabled'},
-                                     "Ticketing is not enabled in the system")
+            if user.is_verified:
```
Comment (iamareebjamal) [inline app/api/events.py]: Remove duplication

File: app/api/events.py [hunk 3]
```diff
@@ -51,6 +51,18 @@
 from app.models.stripe_authorization import StripeAuthorization
 
 
+def check_ticketing(user, modules, data=None):
+    if user.is_verified:
+        raise ForbiddenException({'source': '/data/attributes/is-ticketing-enabled'},
+                                 "Please enter the location of the event")
```
Comment (iamareebjamal) [inline app/api/events.py]: If user is not verified, throw that error and then check properly if location is not defined. What has user being verified got to do with location of the event?

Also all attribute keys are wrong, why are they 'data/attributes/is-ticketing-enabled'?

File: app/api/events.py [hunk 4]
```diff
@@ -117,9 +129,8 @@ def before_post(self, args, kwargs, data=None):
         """
         user = User.query.filter_by(id=kwargs['user_id']).first()
         modules = Module.query.first()
-        if data.get('is_ticketing_enabled', False) and not modules.ticket_include:
-            raise ForbiddenException({'source': '/data/attributes/is-ticketing-enabled'},
-                                     "Ticketing is not enabled in the system")
+        if data.get('is_ticketing_enabled', False):
```
Comment (iamareebjamal) [inline app/api/events.py]: It shouldn't be checked 'if ticketing is enabled'. It should be checked anyway

File: app/api/events.py [hunk 5]
```diff
@@ -452,10 +462,10 @@ def before_patch(self, args, kwargs, data=None):
         :param data:
         :return:
         """
+        user = User.query.filter_by(id=kwargs['user_id']).first()
         modules = Module.query.first()
-        if data.get('is_ticketing_enabled', False) and not modules.ticket_include:
-            raise ForbiddenException({'source': '/data/attributes/is-ticketing-enabled'},
-                                     "Ticketing is not enabled in the system")
+        if data.get('is_ticketing_enabled', False):
```
Comment (iamareebjamal) [inline app/api/events.py]: Same here, and rename the function from `check_ticketing`

File: app/api/events.py [hunk 6]
```diff
@@ -51,6 +51,18 @@
...
 
+def check_ticketing(user, modules):
+    if not user.is_verified:
+        raise ForbiddenException({'source': '/data/attributes/is-ticketing-enabled'},
+                                 "Please verify your Email")
+    elif not modules.ticket_include:
+        raise ForbiddenException({'source': '/data/attributes/is-ticketing-enabled'},
+                                 "Ticket modules not included.")
+    else:
+        raise ForbiddenException({'source': '/data/attributes/is-ticketing-enabled'},
```
Comment (iamareebjamal) [inline app/api/events.py]: Sources are still wrong

File: app/api/events.py [hunk 7]
```diff
@@ -53,13 +53,13 @@
...
-        raise ForbiddenException({'source': '/data/attributes/is-ticketing-enabled'},
+        raise ForbiddenException({'source': '/data/attributes/is-verified'},
                                  "Please verify your Email")
     elif not modules.ticket_include:
-        raise ForbiddenException({'source': '/data/attributes/is-ticketing-enabled'},
+        raise ForbiddenException({'source': '/data/attributes/ticket-include'},
                                  "Ticket modules not included.")
     else:
-        raise ForbiddenException({'source': '/data/attributes/is-ticketing-enabled'},
+        raise ForbiddenException({'source': '/data/attributes/location-name'},
```
Comment (iamareebjamal) [inline app/api/events.py]: This will run unconditionally

Comment (prateekj117) [reply app/api/events.py]: @iamareebjamal What source should I specify in that case. Ticketing is not enabled according to location and user-verifed condition only AFAI could find.

Comment (iamareebjamal) [reply app/api/events.py]: This is not about source. As you can see, the build has failed. You'll see the reason it failed is this line

Comment (prateekj117) [reply app/api/events.py]: @iamareebjamal As the location is been checked in the later stages of the before_post function, I think we should just discard the else statement.

Comment (iamareebjamal) [reply app/api/events.py]: If location is checked, then OK

**Comment by Haider8** [comment]
@prateekj117 What will the error message if ```super-admin``` tries to ```Save draft``` with only ```event name``` ?

**Comment by prateekj117** [comment]
@Haider8 I don't think we need to handle that case as this script is almost the same with some more descriptive messages. @iamareebjamal Your views on this?

**Comment by iamareebjamal** [comment]
Expected Scenarios:

User is unverified: Show unverified error unconditionally. This should be handled by a decorator, but for now, let it be handled as it is

User is verified:
  Event request has tickets and tickets module is disabled: Show ticketing is disabled
  Event request does not have tickets or it has and ticketing module is enabled:
    Event request does not have location: Show location error

Nested conditions mean the above condition is pre-requisite

These flows are common and should be obvious. I don't know why the changes were made so that this message is shown to unverified user but some these validations could be handled by marshmallow. But since it has been days since the issue has been reported, let's close this PR ASAP

File: app/api/events.py [hunk 8]
```diff
@@ -51,13 +51,14 @@
...
-def check_ticketing(user, modules):
+def check_ticketing(user, modules, data=None):
     if not user.is_verified:
         raise ForbiddenException({'source': '/data/attributes/is-verified'},
                                  "Please verify your Email")
-    elif not modules.ticket_include:
-        raise ForbiddenException({'source': '/data/attributes/ticket-include'},
-                                 "Ticket modules not included.")
+    else:
+        if data.get('is_ticketing_enabled', False) and not modules.ticket_include:
```
Comment (iamareebjamal) [inline app/api/events.py]: Should be elif

File: app/api/events.py [hunk 9]
```diff
@@ -51,13 +51,14 @@
 from app.models.stripe_authorization import StripeAuthorization
 
 
-def check_ticketing(user, modules):
+def check_ticketing(user, modules, data=None):
     if not user.is_verified:
         raise ForbiddenException({'source': '/data/attributes/is-verified'},
```
Comment (iamareebjamal) [inline app/api/events.py]: Source is wrong, it should be empty

File: app/api/events.py (hunk 8 above)
Comment (iamareebjamal) [inline app/api/events.py]: The condition is wrong, it should check `data.get('is_ticketing_enabled', False)`

Please check logic before pushing

File: app/api/events.py [hunk 10]
```diff
@@ -461,9 +462,9 @@ def before_patch(self, args, kwargs, data=None):
         """
         user = User.query.filter_by(id=kwargs['user_id']).first()
         modules = Module.query.first()
-        check_ticketing(user, modules)
+        check_ticketing(user, modules, data)
         if data.get('can_pay_by_paypal', False) or data.get('can_pay_by_cheque', False) or \
-                data.get('can_pay_by_bank', False) or data.get('can_pay_by_stripe', False):
+            data.get('can_pay_by_bank', False) or data.get('can_pay_by_stripe', False):
```
Comment (iamareebjamal) [inline app/api/events.py]: revert

File: app/api/events.py [hunk 11]
```diff
@@ -53,10 +53,10 @@
 
 def check_ticketing(user, modules, data=None):
     if not user.is_verified:
-        raise ForbiddenException({'source': '/data/attributes/is-verified'},
+        raise ForbiddenException({'source': ''},
                                  "Please verify your Email")
-    else:
-        if data.get('is_ticketing_enabled', False) and not modules.ticket_include:
+    elif user.is_verified:
+        if data.get('is_ticketing_enabled', True) and not modules.ticket_include:
```
Comment (iamareebjamal) [inline app/api/events.py]: Wrong, the if condition should be in elif

```python
if something:
  ...
else:
  if something_else:
    ...

```

is useless and same as

```python
if something:
  ...
elif something_else:
  ...

```

**Comment by prateekj117** [comment]
@iamareebjamal Please review.

File: app/api/events.py [hunk 12]
```diff
@@ -51,6 +51,15 @@
 from app.models.stripe_authorization import StripeAuthorization
 
 
+def check_ticketing(user, modules, data=None):
```
Comment (iamareebjamal) [inline app/api/events.py]: data is unused. Remove

Comment (iamareebjamal) [inline app/api/events.py]: data is compulsory, remove data=None

Comment (iamareebjamal) [inline app/api/events.py]: Rename appropriately, this is not just checking ticketing

File: app/api/events.py [hunk 13]
```diff
@@ -126,7 +126,7 @@ def before_post(self, args, kwargs, data=None):
         """
         user = User.query.filter_by(id=kwargs['user_id']).first()
         modules = Module.query.first()
-        check_ticketing(user, modules, data)
+        check_create_event(user, modules, data)
         if data.get('can_pay_by_paypal', False) or data.get('can_pay_by_cheque', False) or \
```
Comment (iamareebjamal) [inline app/api/events.py]: Move all common conditions to the function

File: app/api/events.py [hunk 14]
```diff
@@ -461,7 +461,7 @@ def before_patch(self, args, kwargs, data=None):
         """
         user = User.query.filter_by(id=current_identity.id).one()
         modules = Module.query.first()
-        check_ticketing(user, modules, data)
+        check_create_event(user, modules, data)
         if data.get('can_pay_by_paypal', False) or data.get('can_pay_by_cheque', False) or \
```
Comment (iamareebjamal) [inline app/api/events.py]: Like this one

File: app/api/events.py [hunk 15]
```diff
@@ -51,7 +51,7 @@
 from app.models.stripe_authorization import StripeAuthorization
 
 
-def check_ticketing(user, modules, data=None):
+def check_create_event(user, modules, data):
```
Comment (iamareebjamal) [inline app/api/events.py]: Rename to `validate_event`

**Comment by prateekj117** [comment]
@CosmicCoder96 Please review and merge.

**Pull request merged**